eth/catalyst/api.go

```diff
@@ -756,3 +758,50 @@ func (api *ConsensusAPI) heartbeat() {
 func (api *ConsensusAPI) ExchangeCapabilities([]string) []string {
 	return caps
 }
+
+// GetPayloadBodiesV1 implements engine_getPayloadBodiesByHashV1 which allows for retrieval of a list
+// of block bodies by the engine api.
+func (api *ConsensusAPI) GetPayloadBodiesByHashV1(hashes []common.Hash) []*beacon.ExecutionPayloadBodyV1 {
+	var bodies = make([]*beacon.ExecutionPayloadBodyV1, len(hashes))
+	for i, hash := range hashes {
+		block := api.eth.BlockChain().GetBlockByHash(hash)
+		bodies[i] = getBody(block)
+	}
+	return bodies
+}
+
+// GetPayloadBodiesByRangeV1 implements engine_getPayloadBodiesByRangeV1 which allows for retrieval of a range
+// of block bodies by the engine api.
+func (api *ConsensusAPI) GetPayloadBodiesByRangeV1(start, count uint64) []*beacon.ExecutionPayloadBodyV1 {
+	if api.eth.BlockChain().CurrentBlock().NumberU64() < start {
+		// Return [] if the requested range is past our latest block
+		return []*beacon.ExecutionPayloadBodyV1{}
+	}
+	bodies := make([]*beacon.ExecutionPayloadBodyV1, count)
+	for i := uint64(0); i < count; i++ {
```

Comment:
```suggestion
        end := start + count
        last := api.eth.BlockChain().CurrentBlock().NumberU64()
        if end > last {
            end = last
        }
        var bodies []*beacon.ExecutionPayloadBodyV1{}
	for i := start; i < end i++ {
		block := api.eth.BlockChain().GetBlockByNumber(start + i)
		bodies = append(bodies, getBody(block))
```
Something like that?

Reply:
I just did exactly the same again^^

Reply:
But this time I didn't commit, so the glory is all yours!